### PR TITLE
GitHub actions

### PR DESCRIPTION
**Pull request by domesticmouse**

### CODE REVIEW

**Comment by domesticmouse** [comment]
Hey @mit-mit can I get you to have a look at this?

**Comment by domesticmouse** [comment]
FYI @RedBrogdon

File: .github/workflows/dart.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      - name: Install dependencies
+        run: sudo apt-get install -y protobuf-compiler redis
+      - name: Report version
+        run: dart --version
+      - name: Install dependencies
+        run: dart pub get
+      - name: Activate pub globals
+        run: dart pub global activate grinder protoc_plugin
+      - name: Check formatting
+        run: dart format --set-exit-if-changed lib test tool
```
Comment (johnpryan) [inline .github/workflows/dart.yml:32]: nit, but the `--output=none` flag prevents the code from being re-written, which... might be faster? I don't think it really matters.

```suggestion
        run: dart format --output=none --set-exit-if-changed .
```

https://github.com/dart-lang/sdk/issues/44023

**Review by johnpryan — APPROVED** [review]
LGTM

**Pull request merged**